channels/offline-channel-js/src/OfflineChannel.ts

```diff
@@ -0,0 +1,614 @@
+                    let channelIds = storageConfig.primaryOnlineChannelId;
+                    let onlineUrl = _endpoint;
+                    if (channelIds && channelIds.length) {
+                        arrForEach(channelIds, (id) => {
+                            let plugin = _self.core.getPlugin<IChannelControls>(id);
+                            let channel = plugin && plugin.plugin;
+                            if (channel) {
+                                _primaryChannelId = id;
+                                if (isFunction(channel.getOfflineSupport)) {
+                                    _offineSupport = channel.getOfflineSupport();
+                                    onlineUrl = isFunction(_offineSupport.getUrl) && _offineSupport.getUrl();
+                                }
+                                return;
+                            }
+                            
+                        });
+                    }
+
+                    _overrideIkey = storageConfig.overrideInstrumentationKey;
+                
+                    let urlConfig: IUrlLocalStorageConfig = _urlCfg;
+                    let curUrl = onlineUrl || coreConfig.endpointUrl;
+
+                    if (curUrl !== _endpoint) {
+                       
+                        let coreRootCtx = _getCoreItemCtx(coreConfig, core, extensions, pluginChain);
+                        let providerContext: ILocalStorageProviderContext = {
+                            itemCtx: coreRootCtx,
+                            storageConfig: storageConfig,
+                            id: _self.id,
+                            endpoint: curUrl
+                        };
+                        let handler = new OfflineBatchHandler(_self.diagLog(), _self._unloadHooks);
+                        handler.initialize(providerContext);
+                        urlConfig = {
+                            iKey: coreConfig.instrumentationKey,
+                            url: curUrl,
+                            minPersistenceCacheLevel: storageConfig.minPersistenceLevel,
+                            coreRootCtx: coreRootCtx,
+                            providerContext: providerContext,
+                            batchHandler: handler
+                        };
+                        let evtsLimit = storageConfig.eventsLimitInMem;
+                        _inMemoBatch = new InMemoryBatch(_self.diagLog(),curUrl, null, evtsLimit);
+                        _inMemoTimerOut = storageConfig.inMemoMaxTime;
+                        let onlineConfig = ctx.getExtCfg<any>(_primaryChannelId, {}) || {};
+                        _convertUndefined = onlineConfig.convertUndefined;
+                        _endpoint = curUrl;
+                        _setRetryTime();
+                        _maxBatchInterval = storageConfig.maxSentBatchInterval;
+                        _maxBatchSize = storageConfig.maxBatchsize;
+                    }
+                    _urlCfg = urlConfig;
```

Comment:
Add todo: this will overwrite any "previous" if the endpoint is changed as you construct a "new" one. Which means all of the previous stuff will either get lost or left behind in memory -- because it will never be cleaned up on teardown.

Reply:
created task 26681188 for this